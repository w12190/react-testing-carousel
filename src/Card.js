import React from "react";
import "./Card.css";

/** Card
 * 
 * Props: 
 * - caption
 * - src
 * - currNum
 * - totalNum
 * 
 * App => Carousel => Card
 */
//CR: destructer props when it's very simple (like one prop here with caption)
function Card(props) {
  return (
    <div className="Card">
      <h4 className="Card-title">{props.caption}</h4>
      <img className="Card-image" src={props.src} alt={props.caption} />
      <small className="Card-small">
        Image {props.currNum} of {props.totalNum}.
      </small>
    </div>
  );
}

export default Card;
