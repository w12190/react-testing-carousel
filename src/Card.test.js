import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import image1 from "./image1.jpg";

it ("renders (without testing)", function (){
  render(<Card
    caption="test_caption"
    src={image1} //this is the image
    currNum={1}
    totalNum={1}
  />)
})

it ("matches snapshot", function (){
  const {container} = render(
  <Card
    caption="test_caption"
    src={image1} //this is the image
    currNum={1}
    totalNum={1}
  />)
  expect(container).toMatchSnapshot()
})

it('creates an accurate card', function(){
  const {container} = render(
  <Card
    caption="test_caption"
    src={image1} //this is the image
    currNum={1}
    totalNum={1}
  />)

  expect(container.querySelector('.Card-image')).toBeInTheDocument()
  expect(container.querySelector('.not-here')).not.toBeInTheDocument()
})