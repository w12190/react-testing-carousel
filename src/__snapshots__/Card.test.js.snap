// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test_caption
    </h4>
    <img
      alt="test_caption"
      class="Card-image"
      src="image1.jpg"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      1
      .
    </small>
  </div>
</div>
`;
